### PR TITLE
Feat/boarding

### PR DESCRIPTION
**Pull request by joaoqueiroz95**
[boarding.webm](https://github.com/joel-gouveia/frs-booking/assets/112548481/17df7e34-f6e1-46b4-8af0-5308859c5d44)

### CODE REVIEW

**Review by mathvaillant — APPROVED** [review]

File: src/components/Input/NumericInput.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import { Input, InputProps } from "./Input";
+
+interface Props extends InputProps {}
+
+export function NumericInput({ onChangeText, ...props }: Props) {
+  const handleChangeTextNumeric = (text: string) => {
+    if (onChangeText) {
+      onChangeText(text.replace(/[^0-9]/g, ""));
+    }
```
Comment (joel-gouveia) [inline src/components/Input/NumericInput.tsx:10]: You can do `onChangeText?.( ... )`

File: src/components/Input/PasswordInput.tsx [hunk 2]
```diff
@@ -24,7 +24,9 @@ export function PasswordInput(props: Props) {
           <Icon
             name={showPassword ? "eye" : "eye-off"}
             size={22}
-            color={props.hasError ? theme.colors.error : theme.colors.primary.contrastText}
+            color={
+              props.hasError ? theme.colors.error.darkBackground : theme.colors.primary.contrastText
```
Comment (joel-gouveia) [inline src/components/Input/PasswordInput.tsx:28]: Please destruct the props in the function definition

File: src/screens/__tests__/Boarding.test.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    const { getByText } = render(<BoardingScreen />);
+
+    await waitFor(() => {
+      expect(getByText(i18n.t("boarding.board"))).toBeTruthy();
+
+      fireEvent.press(getByText(i18n.t("boarding.board")));
+      expect(mockCreateBoarding).toHaveBeenCalled();
+    });
+  });
+});
```
Comment (joel-gouveia) [inline src/screens/__tests__/Boarding.test.tsx:50]: Add a test for the input as well

File: src/screens/Boarding.tsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+            <IonIcon name="qr-code-outline" size={30} color={theme.colors.primary.contrastText} />
+          </HStack>
+        </Button>
+        <Typography size="lg" style={styles.orText}>
+          {t("common.or")}
+        </Typography>
+        <KeyboardAvoidingView>
+          <NumericInput
+            testID="boarding-input"
+            placeholder="Add ticket number manually"
```
Comment (joel-gouveia) [inline src/screens/Boarding.tsx:70]: Should this be a constant with i8?

File: src/screens/Boarding.tsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+  return (
+    <ScreenLayout
+      isScrollable
+      headerProps={{
+        title: "Boarding",
+        icon: <Icon name="sailboat" size={24} color={theme.colors.primary.main} />,
+      }}
+      footerButtons={[
+        {
+          subtitle: "Main Menu",
```
Comment (joel-gouveia) [inline src/screens/Boarding.tsx:46]: probably a i8 thing

File: src/screens/Boarding.tsx [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+        text2: t("boarding.errors.ticket-number-invalid.subtitle"),
+      });
+    }
+  };
+
+  return (
+    <ScreenLayout
+      isScrollable
+      headerProps={{
+        title: "Boarding",
```
Comment (joel-gouveia) [inline src/screens/Boarding.tsx:41]: i8?

File: src/screens/Boarding.tsx (hunk 4 above)
Comment (joaoqueiroz95) [reply src/screens/Boarding.tsx]: You are right, just updated your comments on the last commit

**Review by joel-gouveia — APPROVED** [review]
LGTM!

**Pull request merged**